oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/MetadataQueryEsDAO.java

```diff
@@ -76,17 +78,19 @@ public MetadataQueryEsDAO(
         ElasticSearchClient client,
         StorageModuleElasticsearchConfig config) {
         super(client);
-        this.queryMaxSize = config.getMetadataQueryMaxSize();
+        this.metadataMaxSize = config.getMetadataQueryMaxSize();
+        this.resultWindowMaxSize = config.getResultWindowMaxSize();
         this.scrollingBatchSize = config.getScrollingBatchSize();
         this.layerSize = Layer.values().length;
+        this.queryMaxSize = Math.min(metadataMaxSize, resultWindowMaxSize);
```

Comment:
Why do you have to run `min` between these two configurations? `metadataQueryMaxSize` is already open for your manual setup. I can't see your point of changing this.

Reply:
I change this because this is also limit by the elasticsearch resultWindowMaxSize, if the metadataQueryMaxSize  setting by 20000 over the default limit by elasticserch it will cause an exception. The metadataQueryMaxSize is add to help the user to query  with elasticserch sroll query to avoid the limit 10000 of elasticserch.In order to fix this problem,I did this by not affect the other query,but support the  elasticserch sroll query, this will separate the queryMaxSize and metadataSize,when the metadataSize is litte than the resultWindowMaxSize, will query the data as the user setting, when the user setting the metadataSize over the resultWindowMaxSize, the elastic scroll query will give the metadataSize result as the user want, the other query will also limit by the resultWindowMaxSize and will not cause an exception.

Reply:
> I change this because this is also limit by the elasticsearch resultWindowMaxSize

Why don't you change the value to fix the expectation? The server setup matters too.

Reply:
If it is a hardcode case, then we could fix it. But AFAIK, all these are exposed to you. Make them right, rather than adding more complex logic to cause further confusing.

Reply:
Thanks for dear wu,I just want to help skywalking to become better and offer an effort.I will remember that not to adding more complex logic and further confusion. Thanks a lot.